docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,100 @@
+# ADR 008: Blocksize independent Commitments for Messages
+
+## Changelog
+
+- 03.08.2022: Initial Draft
+
+## Context
+
+Currently, commitments in Celestia are dependent on the size of the block. The following diagram visualizes how this can happen. The visualizations in this ADR are without parity shares/erasure encoding.
+The yellow block in the diagram is a message of length 11 shares. A square of 4x4 results in the following subtree roots: `A B C D`.
+In [`CreateCommit`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L112-166), we calculate the commitment of the subtree roots with [`HashFromByteSlices`](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/crypto/merkle/tree.go#L7-L21). This resulting merkleroot is `Com`.
+A square of 8x8 results in the following subtree roots: `H1 C D`
+[`HashFromByteSlices`](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/crypto/merkle/tree.go#L7-L21) results now in the merkleroot `Com'`, which is not equal to `Com`.
+
+![Old Commitment](./assets/size-dependent-commitment.png)
+
+To have a commitment independent of the block size, you would need to create a merkle tree over subtree roots that are included in every possible block size.
+We agree on producing the commitment as if the message was put into the `minimalBlocksize`. Therefore the subtree roots will be the same and the resulting commitment as well, no matter the block size. *This is because all the subtree roots in the `minimalBlocksize` are also included in bigger block sizes*.
```

Comment:
```suggestion
We agree on producing the commitment as if the message was put into the `minimumSquareSize`. Therefore the subtree roots will be the same and the resulting commitment as well, no matter the square size. *This is because all the subtree roots in the `minimumSquareSize` are also included in bigger square sizes*.
```

Reply:
[optional] it may help future readers if this section states that the diagram below assumes `minimumSquareSize=4`